web/src/components/core/PageOptions.jsx

```diff
@@ -114,20 +114,40 @@ const Item = ({ children, ...props }) => {
  */
 const PageOptions = ({ children }) => {
   const [isOpen, setIsOpen] = useState(false);
+  const dropdownRef = useRef(null);
+
   const onToggle = () => setIsOpen(!isOpen);
   const onSelect = () => setIsOpen(false);
 
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (dropdownRef.current && !dropdownRef.current.contains(event.target)) {
+        setIsOpen(false);
+      }
+    };
+
+    if (isOpen) {
+      document.addEventListener('click', handleClickOutside);
+    }
+
+    return () => {
+      document.removeEventListener('click', handleClickOutside);
+    };
+  }, [isOpen]);
+
   return (
     <PageOptionsSlot>
-      <Dropdown
-        isOpen={isOpen}
-        toggle={<Toggler onClick={onToggle} />}
-        onSelect={onSelect}
-        dropdownItems={Array(children)}
-        position="right"
-        className="page-options"
-        isGrouped
-      />
+      <span ref={dropdownRef}>
```

Comment:
Suggestion: I prefer a `<div>` instead of `<span>`. After all, the `Dropdown` base element is already a `<div>` so it looks more consistent to me.

Additionally, the `<div>` element supports [flow content](https://developer.mozilla.org/en-US/docs/Web/HTML/Content_categories#flow_content) while the `<span>` is limited to [phrasing content](https://developer.mozilla.org/en-US/docs/Web/HTML/Content_categories#phrasing_content).

Reply:
Yeah definitely, `<div>` is a better choice, sorry for missing this.

I've made the commit to PR addressing this.

Reply:
No problem, thanks a lot for addressing it 😃